api/v1beta2/ibmvpccluster_types.go

```diff
@@ -98,26 +110,108 @@ type AdditionalListenerSpec struct {
+	Name *string `json:"name,omitempty"`
+
+	// algorithm defines the load balancing algorithm to use.
+	// +required
+	Algorithm string `json:"algorithm"`
+
+	// protocol defines the protocol to use for the Backend Pool.
+	// +required
+	Protocol string `json:"protocol"`
+
+	// healthDelay defines the seconds to wait between health checks.
+	// +required
+	HealthDelay int64 `json:"healthDelay"`
+
+	// healthRetries defines the max retries for health check.
+	// +required
+	HealthRetries int64 `json:"healthRetries"`
+
+	// healthTimeout defines the seconds to wait for a health check response.
+	// +required
+	HealthTimeout int64 `json:"healthTimeout"`
+
+	// healthType defines the protocol used for health checks.
+	// +required
+	HealthType string `json:"healthType"`
+
+	// healthMonitorURL defines the URL to use for health monitoring.
+	// +optional
+	HealthMonitorURL *string `json:"healthMonitorURL,omitempty"`
+
+	// healthMonitorPort defines the port to perform health monitoring on.
+	// +optional
+	HealthMonitorPort *int64 `json:"healthMonitorPort,omitempty"`
+}
+
+// ImageSpec defines the desired state of the VPC Custom Image resources for the cluster.
+// +kubebuilder:validation:XValidation:rule="(!has(self.cosInstance) && !has(self.cosBucket) && !has(self.cosObject)) || (has(self.cosInstance) && has(self.cosBucket) && has(self.cosObject))",message="if any of cosInstance, cosBucket, or cosObject are specified, all must be specified"
+// +kubebuilder:validation:XValidation:rule="has(self.name) || has(self.crn) || (has(self.cosInstance) && has(self.cosBucket) && has(self.cosObject))",message="an existing image name or crn must be provided, or to create a new image the cos resources must be provided, with or without a name"
+type ImageSpec struct {
+	// name is the name of the desired VPC Custom Image.
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// crn is the IBM Cloud CRN of the existing VPC Custom Image
+	// +optional
+	CRN *string `json:"crn,omitempty"`
+
+	// cosInstance is the name of the IBM Cloud COS Instance containing the source of the image, if necessary.
+	// +optional
+	COSInstance *string `json:"cosInstance,omitempty"`
+
+	// cosBucket is the name of the IBM Cloud COS Bucket containing the source of the image, if necessary.
+	// +optional
+	COSBucket *string `json:"cosBucket,omitempty"`
+
+	// cosBucketRegion is the COS region the bucket is in.
+	// +optional
+	COSBucketRegion *string `json:"cosBucketRegion,omitempty"`
 
+	// cosObject is the name of a IBM Cloud COS Object used as the source of the image, if necessary.
+	// +optional
+	COSObject *string `json:"cosObject,omitempty"`
+
+	// operatingSystem is the Custom Image's Operating System name.
+	// +optional
+	OperatingSystem *string `json:"operatingSystem,omitempty"`
+
+	// resourceGroup is the Resource Group to create the Custom Image in.
+	// +optional
+	ResourceGroup *IBMCloudResourceReference `json:"resourceGroup,omitempty"`
```

Comment:
I think we have ResourceGroup at various place.  Do you think this option is needed, can't we create image in cluster resource group?

Reply:
The ResourceGroup is imbedded within Power code, I do not wish reuse Power dependent code
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/112f9685f7bec8c53e179edbac52450e7c669c14/api/v1beta2/ibmpowervscluster_types.go#L175-L182

The ResourceGroup can be used for Image creation, or to retrieve an existing image. So, the RG doesn't have to match the Cluster RG.